pandas/_libs/tslibs/util.pxd

```diff
@@ -241,12 +241,9 @@ cdef inline const char* get_c_string_buf_and_size(object py_string,
     cdef:
         const char *buf
 
-    if PyUnicode_Check(py_string):
-        buf = PyUnicode_AsUTF8AndSize(py_string, length)
-    else:
-        PyBytes_AsStringAndSize(py_string, <char**>&buf, length)
+    buf = PyUnicode_AsUTF8AndSize(py_string, length)
```

Comment:
Hmm interesting - so I guess this helper function existed in the first place for 2/3 compat? If this is all it is reduced to why not get rid of this function altogether and just replace with `PyUnicode_AsUTF8AndSize`?

Reply:
> so I guess this helper function existed in the first place for 2/3

thats my guess, yah.

> just replace with PyUnicode_AsUTF8AndSize?

for now at least because we need to figure out what to do about the segfault thing; the imlpementation here might end up changing